### DISCUSSION TITLE
is it possible to set the t0h and  t1h

### DISCUSSION THREAD

**Discussion by MrGokilz** [Makuna/NeoPixelBus#797] [General]
Hello Makuna,

I like your Lib very much. Thanks for your good works.
I am using your lib with ESP32 S3. Is it also possible to change the t0h, t0l, t1h and t1l?

Thanks in advance.

Regards,
Gokilz

**Comment by Makuna** [comment]
https://github.com/Makuna/NeoPixelBus/wiki/NeoPixelBus-object

The method defines the timing.  The method is specific to the LED chip.  What chip are you using?

**Reply by MrGokilz** [reply]
Thanks for the info, I already know about it.

I am just trying to use WL-ICLED
T0H = 0.3µs
T0L = 0.9µs
T1L = 0.3µs
T1H = 0.9µS

The timing looks a bit different from the W2812B.

**Reply by Makuna** [reply]
Its more than just timing of the pulse, there is also the total pulse size or often referred to as the data speed in kbps.  
Further, the reset time is needed.  

The WS2811 is close to that.  I would suggest it.
The Gs1903 is dead on, but has a reset time of 40us, which maybe faster than your chips support.

The spec I see states 200us reset needed.  Further, it does state 1.25 total pusle, so I would avoid the GS1903 as its below that by a bit.

**Reply by Makuna** [reply]
You can create an issue to add specific feature for Wlicled, it will be just an alias of the WS2811.

**Reply by MrGokilz** [reply]
can I maybe "only" change this variable in the correct methode
    const static DRAM_ATTR uint32_t RmtBit0 = Item32Val(300, 900);
    const static DRAM_ATTR uint32_t RmtBit1 = Item32Val(900, 300);
    const static DRAM_ATTR uint16_t RmtDurationReset = FromNs(50000); // 50us

**Reply by MrGokilz** [reply]
I just changed the value and then it works. Thanks for your help

**Reply by Makuna** [reply]
If you are maintaining your own custom version, then you can change it how ever you want ;-)

If you want the support to trickle through NeoPixelBus so you don't have to maintain it, then you will need to do more.

NeoPixelBus gets refactored all the time, so maintaining an external version can get time consuming. I bring this up as a big refactor is in the works right now.

**Comment by MrGokilz** [comment]
yeah, that's the problem, if I change it by my self.

If it is possible, maybe you can define a "special" method, where the user can set the timing by him self 🙏🏻

**Reply by Makuna** [reply]
You can create an issue with this request.  But note that few Methods can support this.  On ESP32, RMT and BitBang can while the I2S cannot.  A few other platforms have BitBang support that can try but don't often have the resolution to achieve anything more than using the defaults.

Dynamic reconfig is not a big goal of this library (check the Wiki for why).  

I would rather spend the time exposing them by name.
